packages/scripts/src/migrate-glintrc.ts

```diff
@@ -0,0 +1,172 @@
+  }
+
+  return tryOrElse(
+    (e) => `Could not patch data:\n\t${JSON.stringify(e)}`,
+    () => {
+      const config = evaluate(contents);
+      config.glint = {
+        environment: rc.environment,
+        transforms: {
+          include: rc.include,
+          exclude: rc.exclude,
+        },
+      };
+
+      return patch(contents, config);
+    }
+  );
+}
+
+function settledToResult<T>(
+  settledResults: Array<PromiseSettledResult<Result<T, unknown>>>
+): Array<Result<T, string>>;
+function settledToResult<T>(
+  settledResults: Array<PromiseSettledResult<T>>
+): Array<Result<T, string>> {
+  return settledResults.map((result) =>
+    result.status === 'fulfilled' ? ok(result.value) : err(JSON.stringify(result.reason))
+  );
+}
+
+function neighborTsconfigPath(rcPath: string): string {
+  return path.join(rcPath, '..', 'tsconfig.json');
+}
+
+function splitResults<T, E>(rs: Array<Result<T, E>>): [Array<T>, Array<E>] {
+  return rs.reduce(
+    ([t, e], r) =>
+      r.match({
+        Ok: (value) => [t.concat(value), e],
+        Err: (reason) => [t, e.concat(reason)],
+      }),
+    [[] as T[], [] as E[]]
+  );
+}
+
+function dumpErrors<T, E>([ts, es]: [T[], E[]]): T[] {
+  es.forEach((e) => console.error(e));
+  return ts;
+}
+
+function reportSuccesses(pathPairs: Array<[string, string]>): void {
+  pathPairs.forEach(([rcPath, tsconfigPath]) =>
+    console.log(`Updated ${tsconfigPath} with contents of ${rcPath}`)
+  );
+}
+
+async function main(): Promise<void> {
+  let paths = yargs
+    .scriptName('migrate-glintrc')
+    .usage('$0 <paths>')
+    .positional('paths', {
+      description: 'path to the `.glintrc.yml` file(s) to migrate',
+      array: true,
+    })
+    .demandCommand()
+    .wrap(100)
+    .strict().argv._;
+
+  let parsed = await Promise.allSettled(
+    paths.map((p) =>
+      loadFile(p).then((contents) =>
+        contents
+          .andThen(parseGlintRcFile)
+          .map((parsed) => [p, parsed] as [string, GlintRc])
+          .mapErr((err) => `${p}: ${err}`)
+      )
+    )
+  )
+    .then(settledToResult)
+    .then(splitResults)
+    .then(dumpErrors);
+
+  let patched = await Promise.allSettled(
+    parsed.map(([rcPath, rc]) => {
+      let tsconfigPath = neighborTsconfigPath(rcPath);
+      return loadOrCreateTsconfig(tsconfigPath).then((contents) =>
+        patchTsconfig(contents, rc)
+          .map((patched) => [rcPath, tsconfigPath, patched] as [string, string, string])
+          .mapErr((err) => `${rcPath}: ${err}`)
+      );
+    })
+  )
+    .then(settledToResult)
+    .then(splitResults)
+    .then(dumpErrors);
+
+  await Promise.allSettled(
+    patched.map(([rcPath, tsconfigPath, patched]) =>
+      saveFile(tsconfigPath, patched).then((writeResult) =>
+        writeResult.map(() => [rcPath, tsconfigPath] as [string, string])
+      )
+    )
+  )
+    .then(settledToResult)
+    .then(splitResults)
+    .then(dumpErrors)
+    .then(reportSuccesses);
+}
+
+main();
```

Comment:
We could drop this in favor of top-level `async` if we care to bump to requiring `module: Node16` in our `tsconfig.json`, but that seems… dicey to do at the top level. I'd perhaps be inclined to do it *only for this script*, since it's safe enough for the binary (we already don't support versions which don't support top-level `await`), but we definitely shouldn't for Glint itself, given the viral nature of that setting. If we do, that should probably also go with making this package be `type: "module"`.

Reply:
Given that folks no longer import `@glint/*` packages at runtime (the closest thing is type-only imports from `@glint/template`), I would guess that migrating to ESM and assuming the availability of top-level `await` would probably be a safe thing for us to do, but I agree that doing that or not is probably something we should explore separately from this.

Reply:
I ended up taking a slightly different approach here, in line with your suggestion in [the comment below][comment]:

> Aside from freeing us up to adopt things like `type: 'module'` and other recent-versions-of-Node goodies earlier, it also means we can worry a little less about bringing in extra dependencies for scripting that we wouldn't necessarily want to drag in as part of `@glint/core`.

I left the root settings as they are, and provided new settings for *this package alone*. That made for a *bunch* of niceties.